[mod] add 'Accept-Language' HTTP header to online processores

return42 opened the pull request:
## What does this PR do?

[mod] add 'Accept-Language' HTTP header to online processors

## Why is this change important?

Most engines that support languages (and regions) use the Accept-Language from the WEB browser to build a response that fits to the language (and region).

## How to test this PR locally?

    make run

compare the results of engines that have language support (e.g. google) before and after the patch.  Select languages (regions) like zh-TW and zh-CN.

The results should be slightly improved (at least in my tests and on my IP).

Closes: https://github.com/searxng/searxng/issues/1567
Closes: https://github.com/searxng/searxng/issues/1321

dalf commented on searx/engines/openstreetmap.py:
```diff
@@ -143,8 +143,6 @@ def request(query, params):
     params['headers']['User-Agent'] = searx_useragent()
 
-    accept_language = 'en' if params['language'] == 'all' else params['language']
-    params['headers']['Accept-Language'] = accept_language
```
* in the master branch "all" means "en" in HTTP header.
* in this PR there is no header at all.

==> 
The query `!osm :all tokyo` returns:
* `Tokyo, Japan` in the master branch
* `東京都, 日本` in this PR

Can be fixed with:
```python
if 'Accept-Language' not in params['headers']:
    params['headers']['Accept-Language'] = 'en'
```

return42 replied:
Thanks, forgot to implement the `all` case / I will fix it.

It is OT here but what me often confuse is  the designation "Default language" and the `:all` tag ...

![image](https://user-images.githubusercontent.com/554536/182177172-34248e5c-cd49-45b8-a9a6-85da7ee169bb.png)

dalf commented on searx/engines/wikipedia.py:
```diff
@@ -41,9 +41,6 @@ def request(query, params):
     language = url_lang(params['language'])
     params['url'] = search_url.format(title=quote(query), language=language)
 
-    if params['language'].lower() in language_variants.get(language, []):
```
LGTM 👍 
I see a difference between `!wp 東京都 :zh` and `!wp 東京都 :zh-hk`

dalf commented on searx/engines/duckduckgo_definitions.py:
```diff
@@ -62,7 +62,6 @@ def request(query, params):
     params['url'] = URL.format(query=urlencode({'q': query}))
     language = match_language(params['language'], supported_languages, language_aliases)
     language = language.split('-')[0]
-    params['headers']['Accept-Language'] = language
```
`!ddd paris :fr` returns:
* a result in French in the master branch
* a result in English in this PR. I don't see the "Accept-Language" header in the log.

With `send_accept_language_header = True` it works as before.

dalf commented on searx/settings.yml:
```diff
@@ -322,24 +322,29 @@ engines:
   - name: wikipedia
     engine: wikipedia
     shortcut: wp
+    send_accept_language_header: true
```
Most of the engines have the configuration in the Python code rather than the .yml file
It would be better to add `send_accept_language_header = True` to wikipedia.py (and to other engines).

dalf commented on searx/network/network.py:
```diff
@@ -265,6 +265,7 @@ async def call_client(self, stream, method, url, **kwargs):
         kwargs_clients = Network.extract_kwargs_clients(kwargs)
         while retries >= 0:  # pragma: no cover
             client = await self.get_client(**kwargs_clients)
+            self._logger.debug("HTTP request arguments--> %s", kwargs)
```
Can we discuss about this change in another PR?

* httpx might add to some additional HTTP header
* It makes the log verbose

dalf reviewed (changes requested):
except the few issues, LGTM .

return42 committed:
[doc] add description of method EngineProcessor.get_params()

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[mod] add 'Accept-Language' HTTP header to online processores

Most engines that support languages (and regions) use the Accept-Language from
the WEB browser to build a response that fits to the language (and region).

- add new engine option: send_accept_language_header

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>
